lib/services/role.go

```diff
@@ -1538,6 +1553,11 @@ func (set RoleSet) CheckAccessToSAMLIdP(authPref types.AuthPreference) error {
 		if !options.IDP.SAML.Enabled.Value {
 			return trace.AccessDenied("user has been denied access to the SAML IdP by role %s", role.GetName())
 		}
+
+		if options.RequireMFAType.IsSessionMFARequired() && !state.MFAVerified {
```

Comment:
Don't we need to handle  also the `state.MFARequired == services.MFARequiredNever`  state ?

Reply:
If the role requires MFA then `state.MFARequired` will always be `per-role`, but in case of future logic changes I'll add the check.